app/controllers/pieces_controller.rb

```diff
@@ -10,7 +10,7 @@ def show
   def update
     @piece = Piece.find(params[:id])
     @game = Game.find(@piece.game_id)
-    @piece.update_attributes(piece_params)
+    @piece.move_to!(piece_params[:row_position].to_i, piece_params[:col_position].to_i)
```

Comment:
@takehiromouri @jonathanpike @addstar34 @ronny2205 Actually, the user *always* has a chance to post something (clicking a button in a browser isn't the only way to send a POST, you can use tools like [PostMan](https://www.getpostman.com/) to manually craft requests, and script things with `curl`). Strong params should *always* be used, because people are evil.

But also, because of practice. When you use strong params you not only prevent against some types of attacks, but you prevent yourself from potentially relying on undefined parameters (parameters you post from the frontend, but don't have defined in the strong params), not very relevant in this application, but it's a good habit to get into.

Lastly, why not? Do we gain something from accessing the parameters directly? There aren't any downsides that I can see to using the strong params, so you should be consistent in your usage.